vector/src/main/java/im/vector/app/features/login/qr/QrCodeLoginInstructionsFragment.kt

```diff
@@ -0,0 +1,102 @@
+import im.vector.app.features.qrcode.QrCodeScannerActivity
+import timber.log.Timber
+
+@AndroidEntryPoint
+class QrCodeLoginInstructionsFragment : VectorBaseFragment<FragmentQrCodeLoginInstructionsBinding>() {
+
+    private val viewModel: QrCodeLoginViewModel by activityViewModel()
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentQrCodeLoginInstructionsBinding {
+        return FragmentQrCodeLoginInstructionsBinding.inflate(inflater, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        observeViewState()
+        initScanQrCodeButton()
+        initShowQrCodeButton()
+    }
+
+    private fun observeViewState() {
+        viewModel.onEach {
+            if (it.loginType == QrCodeLoginType.LOGIN) {
+                views.qrCodeLoginInstructionsView.setInstructions(
+                        listOf(
+                                getString(R.string.qr_code_login_new_device_instruction_1),
+                                getString(R.string.qr_code_login_new_device_instruction_2),
+                                getString(R.string.qr_code_login_new_device_instruction_3),
+                        )
+                )
+            } else {
+                views.qrCodeLoginInstructionsView.setInstructions(
+                        listOf(
+                                getString(R.string.qr_code_login_link_a_device_scan_qr_code_instruction_1),
+                                getString(R.string.qr_code_login_link_a_device_scan_qr_code_instruction_2),
+                        )
+                )
+            }
+        }
+    }
+
+    private fun initShowQrCodeButton() {
+        views.qrCodeLoginInstructionsShowQrCodeButton.debouncedClicks {
+            viewModel.handle(QrCodeLoginAction.ShowQrCode)
+        }
+    }
+
+    private fun initScanQrCodeButton() {
+        views.qrCodeLoginInstructionsScanQrCodeButton.debouncedClicks {
+            QrCodeScannerActivity.startForResult(requireActivity(), scanActivityResultLauncher)
+        }
+    }
+
+    private val scanActivityResultLauncher = registerStartForActivityResult { activityResult ->
+        if (activityResult.resultCode == Activity.RESULT_OK) {
+            val scannedQrCode = QrCodeScannerActivity.getResultText(activityResult.data)
+            val wasQrCode = QrCodeScannerActivity.getResultIsQrCode(activityResult.data)
+
+            if (wasQrCode && !scannedQrCode.isNullOrBlank()) {
+                onQrCodeScanned(scannedQrCode)
+            } else {
+                onQrCodeScannerFailed()
+            }
+        }
+    }
+
+    private fun onQrCodeScanned(scannedQrCode: String) {
+        viewModel.handle(QrCodeLoginAction.OnQrCodeScanned(scannedQrCode))
+    }
+
+    private fun onQrCodeScannerFailed() {
+        Timber.d("QrCodeLoginInstructionsFragment.onQrCodeScannerFailed")
```

Comment:
Should we prompt the user about the error in this case?

Reply:
Hugh is implementing it in another PR.